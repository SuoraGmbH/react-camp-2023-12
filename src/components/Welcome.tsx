import React from "react";

interface Props {
  name: string;
}

const Welcome: React.FunctionComponent<Props> = ({ name }) => {
  return <div>Hallo {name}</div>;
};

export default Welcome;
