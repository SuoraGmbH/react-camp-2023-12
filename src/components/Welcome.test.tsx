import { render, screen } from "@testing-library/react";
import Welcome from "./Welcome.tsx";
import { expect } from "vitest";

describe("<Welcome />", () => {
  test("smoke test123", () => {
    render(<Welcome name="Michael" />);
  });

  it("should say Hallo Michael", () => {
    render(<Welcome name="Michael" />);

    expect(screen.getByText("Hallo Michael")).toBeVisible();
  });

  it("should greet the user", () => {
    render(<Welcome name="Michael" />);

    expect(screen.getByText("Hallo", { exact: false })).toBeVisible();
    expect(screen.getByText(/hallo|hola|salut/i)).toBeVisible();
  });

  it("should say the users name", () => {
    render(<Welcome name="Michael" />);

    expect(screen.getByText(/michael/i)).toBeVisible();
  });

  it("should do exactly what it did before!", () => {
    const { container } = render(<Welcome name="Michael" />);

    expect(container).toMatchSnapshot();
  });
});
