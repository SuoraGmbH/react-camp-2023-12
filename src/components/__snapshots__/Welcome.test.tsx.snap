// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Welcome /> > should do exactly what it did before! 1`] = `
<div>
  <div>
    Hallo 
    Michael
  </div>
</div>
`;
